src/main/scala/org/broadinstitute/clio/dataaccess/CachedServerStatusDAO.scala

```diff
@@ -0,0 +1,47 @@
+package org.broadinstitute.clio.dataaccess
+
+import org.broadinstitute.clio.model.ServerStatusInfo
+
+import scala.concurrent.Future
+import scala.util.Try
+
+/**
+  * Stores the server status in a synchronized mutable variable.
+  */
+class CachedServerStatusDAO private() extends ServerStatusDAO {
+  private val mutex = new Object()
+  private var currentStatus: ServerStatusInfo = ServerStatusInfo.NotStarted
+
+  override def setStatus(status: ServerStatusInfo): Future[Unit] = Future.fromTry(
+    Try {
+      mutex synchronized {
+        import CachedServerStatusDAO.level
+        if (level(currentStatus) < level(status))
```

Comment:
It seems like you'd want an abstraction here, e.g. `canTransitionTo()` on `ServerStatusInfo`, in case the rule for transitioning status change or become more complicated.